File: ts/packages/ts/src/ConfigStore.ts
import { type IConfigStore, type ConfigStoreQuery, type Config } from './types';

export abstract class ConfigStore implements IConfigStore {
  constructor(public readonly type: string) {}

  async init() {}
  abstract get(queries: ConfigStoreQuery[]): Promise<Config[]>;
  abstract set(configs: Config[]): Promise<void>;
}

File: py/src/configu/model/config_store.py
from abc import abstractmethod, ABC

from typing import List

from .generated import ConfigStore as IConfigStore, ConfigStoreQuery, Config


class ConfigStore(ABC, IConfigStore):
    def init(self):
        """Anything that needs init before running Commands"""

    @abstractmethod
    def get(self, queries: List[ConfigStoreQuery]) -> List[Config]:
        raise NotImplemented()

    @abstractmethod
    def set(self, configs: List[Config]) -> None:
        raise NotImplemented()
